flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java

```diff
@@ -70,7 +72,11 @@
 	public HiveTableSource(JobConf jobConf, ObjectPath tablePath, CatalogTable catalogTable) {
 		this.jobConf = Preconditions.checkNotNull(jobConf);
 		this.tablePath = Preconditions.checkNotNull(tablePath);
-		this.catalogTable = Preconditions.checkNotNull(catalogTable);
+
+		Preconditions.checkArgument(catalogTable instanceof CatalogTableImpl);
```

Comment:
Why `catalogTable` must be `CatalogTableImpl ` type?
If it is a must, we may change the constructor.

Reply:
`HiveTableSource` is only initialized in `HiveTableFactory::createInputFormatTableSource`, which is only called from `HiveTableFactory::createTableSource`. This method guarantees that `table` is `CatalogTableImpl`.

Reply:
Though this method guarantees that `table` is `CatalogTableImpl`, it is not a must.
It's better using `CatalogTable` here.